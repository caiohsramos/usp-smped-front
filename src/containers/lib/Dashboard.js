import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from './Container';
import { DashboardController } from '../../controllers';
import { Button } from '../../atoms';
import { Table } from '../../components';

class Dashboard extends Container {
	constructor(props) {
		super(props);

		this.state = {
			expandedStatus: {},
			formsMock: [{name: 'Form1', id:'r982ur982', version: '1.0', author: 'Marcos', description: 'Lorem ipsum condimentum sociosqu lorem auctor tellus praesent donec leo arcu, mollis ipsum eu nisi leo.'},
                                    {name: 'Form2', id:'rdsadr982', version: '2.0', author: 'Aline', description: 'Lorem ipsum lorem fames maecenas sollicitudin nullam mattis netus, ligula convallis mauris rhoncus nisi.'},
                                    {name: 'Form3', id:'rdsdasd22', version: '3.0', author: 'Gabrielly', description: 'Vehicula morbi a fermentum tincidunt etiam curabitur nisi dictumst, aenean malesuada suscipit porttitor.'},
                                    {name: 'Form4', id:'rdsdda323', version: '4.0', author: 'Lucas', description: 'Dolor id nisl etiam potenti aptent duis tortor ultrices ornare, convallis fermentum magna malesuada cubilia.'}]
		};

		const toController = {
			callback: this.callback,
			getState: this.getState,
			getProps: this.getProps,
			router: props.router
		};
		this.controller = new DashboardController(toController);
	}

	render() {
		const { handleClick, handleTest, handleExpandAction } = this.controller;
		const { formsMock, expandedStatus } = this.state;

		return (
            <div className='new-form'>
                <Table
                        data={formsMock}
                        handleClick={handleTest}
                        expandedState={expandedStatus}
                        handleExpandClick={handleExpandAction}/>
                    <Button
                        id='button'
                        type='button'
                        label='Novo formulÃ¡rio'
                        clickAction={handleClick}
                    />
            </div>
		);
	}
}

const mstp = state => {
	return {};
};


export default connect(mstp, {})(Dashboard);
