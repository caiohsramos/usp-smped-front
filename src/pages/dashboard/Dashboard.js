import React from 'react';
import { connect } from 'react-redux';
import { Container } from '../Container';
import CenteredTabs from './components/Tabs';
import { DashboardController } from './DashboardController';
import { Button } from '../../common';
import { Table } from './components/Table';
import { clearMsg } from '../../actions/MessageActions';
import Snackbar from '@material-ui/core/Snackbar';
import { bindActionCreators } from 'redux';

class Dashboard extends Container {
    constructor(props) {
        super(props);
        this.state = {
            formsList: [],
            tabState: 0
        };
        const toController = {
            callback: this.callback,
            getState: this.getState,
            getProps: this.getProps,
            router: props.router
        };
        this.controller = new DashboardController(toController);
        this.controller.fetchForms();
    }

    getAxes(formList) {
        let allAxis = formList.map((form) => form.activity);
        return allAxis.filter((axisName, idx) => allAxis.indexOf(axisName) === idx && axisName !== "");
    }

    render() {
        const { handleClick, handleFormView } = this.controller;
        const { formsList, tabState } = this.state;
        const axisList = this.getAxes(formsList);
        return (
            <div className='new-form'>
                <Snackbar
                    anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
                    open={this.props.message.success}
                    onClose={this.props.clearMsg}
                    className={this.props.message.success != '' ? this.props.message.success ? 'snack-success' : 'snack-fail' : 'snack-success'}
                    ContentProps={{
                        'aria-describedby': 'message-id',
                    }}
                    message={<span id="message-id">{this.props.message.msg}</span>}
                />
                <h1>Relatórios de atividades</h1>
                <CenteredTabs
                    tabState={tabState}
                    handleChangeTab={(tabState) => this.callback({ ...this.state, ...{ tabState } })}
                />
                <Table
                    data={formsList}
                    axisList={axisList}
                    filterType={tabState}
                    handleFormView={handleFormView}
                />
                <Button
                    id='button'
                    type='button'
                    label='Novo formulário'
                    clickAction={handleClick}
                />
            </div>
        );
    }
}

const mstp = state => {
    return {
        message: state.message
    };
};

const mdtp = dispatch =>
    bindActionCreators({ clearMsg }, dispatch);

export default connect(mstp, mdtp)(Dashboard);