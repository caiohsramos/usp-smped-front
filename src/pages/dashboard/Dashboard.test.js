import React from 'react';
import Dashboard from './Dashboard';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import reducer from '../../reducers';

const store = createStore(reducer);

it('renders correctly', () => {
    const tree = renderer
        .create(<Dashboard store={store} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
