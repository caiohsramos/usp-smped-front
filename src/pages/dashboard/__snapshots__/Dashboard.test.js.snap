// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="new-form"
>
  <h1>
    Relatórios de atividades
  </h1>
  <div
    className="MuiTabs-root-8"
  >
    <div
      className="MuiTabs-flexContainer-9"
    >
      <div
        className="MuiTabs-scroller-11 MuiTabs-fixed-12"
        onScroll={[Function]}
        role="tablist"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="MuiTabs-flexContainer-9 MuiTabs-centered-10"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root-29 MuiTab-root-17 MuiTab-textColorPrimary-20 MuiTab-selected-22"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTab-wrapper-25"
            >
              <span
                className="MuiTab-labelContainer-26"
              >
                <span
                  className="MuiTab-label-27"
                >
                  Secretarias
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-91"
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root-29 MuiTab-root-17 MuiTab-textColorPrimary-20"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTab-wrapper-25"
            >
              <span
                className="MuiTab-labelContainer-26"
              >
                <span
                  className="MuiTab-label-27"
                >
                  Eixos
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-91"
            />
          </button>
        </div>
        <span
          className="MuiPrivateTabIndicator-root-32 MuiPrivateTabIndicator-colorPrimary-33 MuiTabs-indicator-16"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="table"
    role="list"
  >
    <div>
      <div>
        <div
          className="MuiPaper-root-38 MuiPaper-elevation1-41 MuiExpansionPanel-root-35 category"
        >
          <div
            aria-expanded={false}
            className="MuiButtonBase-root-29 MuiExpansionPanelSummary-root-65"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiExpansionPanelSummary-content-69"
            >
              Secretaria Municipal de Direitos Humanos
            </div>
            <div
              aria-hidden="true"
              className="MuiButtonBase-root-29 MuiIconButton-root-71 MuiExpansionPanelSummary-expandIcon-70"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={-1}
            >
              <span
                className="MuiIconButton-label-76"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-77"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-91"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            className="MuiCollapse-container-86"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper-88"
            >
              <div
                className="MuiCollapse-wrapperInner-89"
              >
                <div
                  className="MuiExpansionPanelDetails-root-90 category-details"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="MuiPaper-root-38 MuiPaper-elevation1-41 MuiExpansionPanel-root-35 category"
        >
          <div
            aria-expanded={false}
            className="MuiButtonBase-root-29 MuiExpansionPanelSummary-root-65"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiExpansionPanelSummary-content-69"
            >
              Secretaria Municipal da Pessoa com Deficiência
            </div>
            <div
              aria-hidden="true"
              className="MuiButtonBase-root-29 MuiIconButton-root-71 MuiExpansionPanelSummary-expandIcon-70"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={-1}
            >
              <span
                className="MuiIconButton-label-76"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-77"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-91"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            className="MuiCollapse-container-86"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper-88"
            >
              <div
                className="MuiCollapse-wrapperInner-89"
              >
                <div
                  className="MuiExpansionPanelDetails-root-90 category-details"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="btn btn-lg btn-primary btn-block"
    id="button"
    onClick={[Function]}
    type="button"
  >
     
    Novo formulário
     
  </button>
</div>
`;
